cargo-creusot/src/why3_launcher.rs

```diff
@@ -0,0 +1,124 @@
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use super::helpers::Result;
+use anyhow::anyhow;
+use include_dir::{include_dir, Dir};
+
+static PRELUDE: Dir<'static> = include_dir!("$CARGO_MANIFEST_DIR/../prelude");
+
+#[derive(Copy, Clone, Debug)]
+pub enum Why3Mode {
+    Ide,
+}
+
+impl std::fmt::Display for Why3Mode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Why3Mode::Ide => f.write_str("ide"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Why3Launcher {
+    mode: Why3Mode,
+    why3_path: Option<PathBuf>,
+    config_file: Option<PathBuf>,
+    args: Option<String>,
+    output_file: Option<PathBuf>,
+}
+
+impl Why3Launcher {
+    pub fn new(
+        mode: Why3Mode,
+        why3_path: Option<PathBuf>,
+        config_file: Option<PathBuf>,
+        args: Option<String>,
+        output_file: Option<PathBuf>,
+    ) -> Self {
+        Self { mode, why3_path, config_file, args, output_file }
+    }
+
+    pub fn make(&self, temp_dir: &Path) -> Result<Command> {
+        let Some(ref mlcfg_file) = self.output_file else {
+            return Err(anyhow!("can't launch why3, no mlcfg_file specify"));
+        };
+        let mode = self.mode.to_string();
+        PRELUDE.extract(temp_dir).expect("can't launch why3, could extract prelude into temp dir");
```

Comment:
Not that I have any objection with this, but why is it now necessary to copy the prelude to a temporary directory? Just hygiene? Or is there a more fundamental reason?

Reply:
I took this code in: creusot/src/run_why3.rs::run_why3